wolfcrypt/src/aes.c

```diff
@@ -8884,23 +8887,25 @@ static WARN_UNUSED_RESULT int wc_AesGcmDecrypt_STM32(
      * For TLS blocks the authTag is after the output buffer, so save it */
     XMEMCPY(tagExpected, authTag, authTagSz);
 
-    /* Authentication buffer - must be 4-byte multiple zero padded */
-    authPadSz = authInSz % sizeof(word32);
-    if (authPadSz != 0) {
-        authPadSz = authInSz + sizeof(word32) - authPadSz;
-    }
-    else {
-        authPadSz = authInSz;
-    }
-
+    /* Authentication buffer */
+#if STM_CRYPT_HEADER_WIDTH == 1
+    authPadSz = 0; /* CubeHAL supports byte mode */
+#else
+    authPadSz = authInSz % STM_CRYPT_HEADER_WIDTH;
+#endif
 #ifdef WOLFSSL_STM32MP13
     /* STM32MP13 HAL at least v1.2 and lower has a bug with which it needs a
-     * minimum of 16 bytes for the auth
-     */
+     * minimum of 16 bytes for the auth */
     if ((authInSz > 0) && (authInSz < 16)) {
         authPadSz = 16 - authInSz;
     }
 #endif
+    if (authPadSz != 0) {
+        authPadSz = authInSz + STM_CRYPT_HEADER_WIDTH - authPadSz;
+    }
+    else {
+        authPadSz = authInSz;
+    }
```

Comment:
This breaks STM32MP1. Can you please make this if/else the `#else` for the block above it? Then everything works. So:

```c
#ifdef WOLFSSL_STM32MP13
    /* STM32MP13 HAL at least v1.2 and lower has a bug with which it needs a
     * minimum of 16 bytes for the auth */
    if ((authInSz > 0) && (authInSz < 16)) {
        authPadSz = 16 - authInSz;
    }
#else
    if (authPadSz != 0) {
        authPadSz = authInSz + STM_CRYPT_HEADER_WIDTH - authPadSz;
    }
    else {
        authPadSz = authInSz;
    }
#endif
```